### PR TITLE
`GatewayClassObservedGenerationBump` as a core feature

### LINKED ISSUE #1780: `GatewayClassObservedGenerationBump` feature type
**What happened**:

`GatewayClassObservedGenerationBump` is marked as an additional feature, but it Is marked neither as extended conformance nor as experimental conformance: https://github.com/kubernetes-sigs/gateway-api/blob/f6d4eb7a31506be144d5288526eb5f3cceb1bcf1/conformance/utils/suite/suite.go#L56-L57
My guess is that it is extended conformance, is it correct? If so, we should add `(extended conformance)` at the end of the comment line as all the other features above and below.

### PR DESCRIPTION
**Pull request by mlavacca**
**What type of PR is this?**

**What this PR does / why we need it**:

The additional feature `GatewayClassObservedGenerationBump` has been removed, and the test is now executed as a core one.

**Which issue(s) this PR fixes**:

Fixes #1780 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by shaneutt — APPROVED** [review]
Indeed, thank you @mlavacca!

This matches the original intention as I understood it, _however_ let's make sure we let a few people take a look before we merge in case there were different takes (we wouldn't want to break someone).

**Review by LiorLieberman — APPROVED** [review]
Thanks @mlavacca ! 
This LGTM, but we do not have any way to check who does not conform with it yet. 
So perhaps we should ask in Slack if someone is not conforming with this?

**Review by arkodg — APPROVED** [review]

File: conformance/utils/suite/suite.go
Comment (dprotaso) [inline conformance/utils/suite/suite.go:57]: Maybe mark this `Deprecated` for a single release and then drop it in a subsequent one

**Comment by shaneutt** [comment]
We've received some feedback in Slack that at least 1 implementation would like some time to make some adjustments for this. I think we should probably put this on the agenda for next weeks meeting so we can check in with others too.

**Comment by robscott** [comment]
We've talked about this in some meetings without actually leaving an update on the PR. I think we're in agreement that this should _eventually_ happen, but this test makes a relatively large change, it requires implementations to support GatewayClasses with arbitrary names. I believe there are at least some in-cluster implementations of the API that are deployed with a hardcoded GatewayClass name. 

The intent of the API is for every implementation to be able to support GatewayClasses based on the controller name, not just the name of the class. We have not clearly specified that anywhere yet though. We likely need to first clarify that that is a requirement (ideally by v0.7.0) and then make this specific change in a follow up release.

**Comment by robscott** [comment]
Filed https://github.com/kubernetes-sigs/gateway-api/issues/1822 as a blocker for this PR.

**Comment by shaneutt** [comment]
As it stands we're blocked on https://github.com/kubernetes-sigs/gateway-api/issues/1822 for this PR, so I'm going to mark it as `draft` for now just to indicate that it doesn't need active review (and we can mark it ready again whenever we're ready).

**Comment by shaneutt** [comment]
#2228 added some language in support of this behavior to indicate that implementations MUST manage the status for any `GatewayClass` which carries their `controllerName`.

**Comment by mlavacca** [comment]
I've revamped this one and put it in Ready for review as it has been unblocked.

/cc @shaneutt

**Comment by robscott** [comment]
Thanks @mlavacca!

/lgtm

**Pull request merged**
